project/Build.scala

```diff
@@ -803,6 +809,19 @@ object Build {
       javaOptions := (javaOptions in `dotty-compiler-bootstrapped`).value
     )
 
+  lazy val `dotty-tasty-consumer` = project.in(file("tasty-consumer")).
+    withCommonSettings(Bootstrapped).
+    // We want the compiler to be present in the compiler classpath when compiling this project but not
+    // when compiling a project that depends on dotty-staging (see sbt-dotty/sbt-test/sbt-dotty/tasty-consumer-example-project),
+    // but we always need it to be present on the JVM classpath at runtime.
+    dependsOn(dottyCompiler(Bootstrapped) % "provided").
+    dependsOn(dottyCompiler(Bootstrapped) % "compile->runtime").
```

Comment:
I can't remember why we use multiple lines like this in staging instead of `provided; compiler->runtime; test->test`

Reply:
Something did not work with that. SBT rejected it at the time, but maybe we can make it work now that we upgraded.

The same should be changed for `dotty-staging`.